src/AzureExtension/DevBox/DevBoxProvider.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft Corporation and Contributors
+// Licensed under the MIT license.
+
+using System.Text.Json;
+using AzureExtension.Contracts;
+using DevHomeAzureExtension.DataModel;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Windows.DevHome.SDK;
+using Windows.Foundation;
+
+namespace AzureExtension.DevBox;
+
+public class DevBoxProvider : IComputeSystemProvider, IDisposable
+{
+    private readonly IHost _host;
+
+    public DevBoxProvider(IHost host)
+    {
+        _host = host;
+    }
+
+    string IComputeSystemProvider.DefaultComputeSystemProperties
+    {
+        get => throw new NotImplementedException();
+        set => throw new NotImplementedException();
+    }
+
+    public string DisplayName => "DevBox Provider";
+
+    public string Id => "182AF84F-D5E1-469C-9742-536EFEA94630";
+
+    public string Properties => throw new NotImplementedException();
+
+    public ComputeSystemProviderOperation SupportedOperations => throw new NotImplementedException();
+
+    private bool IsValid(JsonElement jsonElement)
+    {
+        return jsonElement.ValueKind != JsonValueKind.Undefined;
+    }
+
+    public IEnumerable<IComputeSystem> GetComputeSystemsAsync(IDeveloperId? developerId)
+    {
+        var computeSystems = new List<IComputeSystem>();
+
+        var mgmtSvc = _host.Services.GetService<IDevBoxManagementService>();
+        if (mgmtSvc != null)
+        {
+            return computeSystems;
+        }
+        else
+        {
+            Log.Logger()?.ReportError($"Error getting systems: Rest Service not configured");
+            throw new ArgumentException($"Rest Service needs to be configured.");
+        }
+    }
+
+    public IAsyncOperation<CreateComputeSystemResult> CreateComputeSystemAsync(string options) => throw new NotImplementedException();
+
+    public IAsyncOperation<ComputeSystemsResult> GetComputeSystemsAsync(IDeveloperId developerId, string options)
+    {
+        return Task.Run(() =>
+        {
+            var computeSystems = GetComputeSystemsAsync(developerId);
+            return new ComputeSystemsResult(computeSystems);
+        }).AsAsyncOperation();
+    }
+
+    public void Dispose()
+    {
+        GC.SuppressFinalize(this);
+    }

```

Comment:
Do we need IDisposable? There is nothing to dispose now.
If we actually need it there is specific pattern to implement IDisposable: https://learn.microsoft.com/en-us/dotnet/api/system.idisposable?view=net-8.0

Reply:
Technically, you are correct.  One of the properties added requires adding this.  I don't know which one though.